### PR TITLE
imp: address code review comments

### PR DESCRIPTION
**Pull request by grondo**
@dun left some good comments on #150, but the PR was already merged. This PR addresses those comments.

### CODE REVIEW

File: src/imp/exec/exec.c [hunk 1]
```diff
@@ -207,7 +207,8 @@ static void __attribute__((noreturn)) imp_exec (struct imp_exec *exec)
 
 static void fwd_signal (int signal)
 {
-    kill (imp_child, signal);
+    if (imp_child > 0)
+        kill (imp_child, signal);
```
Comment (garlick) [inline src/imp/exec/exec.c:211]: Right, protect self, make sure only to kill children.  That sounds like good behavior.

**Review by dun — APPROVED** [review]
:+1:

**Review by garlick — APPROVED** [review]
FWIW these changes all look good to me, and appear to track with @dun's review comments.

**Comment by grondo** [comment]
Thanks @dun and @garlick!

**Pull request merged**